baseband/mark5b/base.py

```diff
@@ -99,22 +99,20 @@ def get_frame_rate(self):
         try:
             return super(Mark5BFileReader, self).get_frame_rate()
         except Exception as exc:
-            oldpos = self.tell()
-            self.seek(0)
-            try:
-                header0 = self.read_header()
-                self.seek(header0.payload_nbytes, 1)
-                header1 = self.read_header()
-                tdelta = header1.fraction - header0.fraction
-                if tdelta == 0.:
-                    exc.args += ("frame rate can also not be determined "
-                                 "from the first two headers, as they have "
-                                 "identical fractional seconds.",)
-                return np.round(1 / tdelta) * u.Hz
-            except Exception:
-                pass
-            finally:
-                self.seek(oldpos)
+            with self.temporary_offset():
+                try:
+                    self.seek(0)
+                    header0 = self.read_header()
+                    self.seek(header0.payload_nbytes, 1)
+                    header1 = self.read_header()
+                    tdelta = header1.fraction - header0.fraction
+                    if tdelta == 0.:
```

Comment:
Know this was there before, but why are we adding to exception arguments here? The exception will just be ignored anyway...

Reply:
Ah, never mind, I now see again why this is here!